`read_commandline` supports polars engine

samukweku opened the pull request:
# PR Description

Please describe the changes proposed in the pull request:

- `read_commandline` now supports polars

<!-- Please also identify below which issue that has been raised that you are going to close. -->

**This PR relates to #1352 **


```py
import janitor as jn
jn.read_commandline('unzip -p ../Downloads/iris.zip', engine='polars')
Out[12]: 
shape: (150, 5)
┌──────────────┬─────────────┬──────────────┬─────────────┬───────────┐
│ sepal_length ┆ sepal_width ┆ petal_length ┆ petal_width ┆ species   │
│ ---          ┆ ---         ┆ ---          ┆ ---         ┆ ---       │
│ f64          ┆ f64         ┆ f64          ┆ f64         ┆ str       │
╞══════════════╪═════════════╪══════════════╪═════════════╪═══════════╡
│ 5.1          ┆ 3.5         ┆ 1.4          ┆ 0.2         ┆ setosa    │
│ 4.9          ┆ 3.0         ┆ 1.4          ┆ 0.2         ┆ setosa    │
│ 4.7          ┆ 3.2         ┆ 1.3          ┆ 0.2         ┆ setosa    │
│ 4.6          ┆ 3.1         ┆ 1.5          ┆ 0.2         ┆ setosa    │
│ 5.0          ┆ 3.6         ┆ 1.4          ┆ 0.2         ┆ setosa    │
│ …            ┆ …           ┆ …            ┆ …           ┆ …         │
│ 6.7          ┆ 3.0         ┆ 5.2          ┆ 2.3         ┆ virginica │
│ 6.3          ┆ 2.5         ┆ 5.0          ┆ 1.9         ┆ virginica │
│ 6.5          ┆ 3.0         ┆ 5.2          ┆ 2.0         ┆ virginica │
│ 6.2          ┆ 3.4         ┆ 5.4          ┆ 2.3         ┆ virginica │
│ 5.9          ┆ 3.0         ┆ 5.1          ┆ 1.8         ┆ virginica │
└──────────────┴─────────────┴──────────────┴─────────────┴───────────┘
```

<!-- As you go down the PR template, please feel free to delete sections that are irrelevant. -->

# PR Checklist

<!-- This checklist exists for newcomers who are not yet familiar with our requirements. If you are experienced with
the project, please feel free to delete this section. -->

Please ensure that you have done the following:

1. [x] PR in from a fork off your branch. Do not PR from `<your_username>`:`dev`, but rather from `<your_username>`:`<feature-branch_name>`.
<!-- Doing this helps us keep the commit history much cleaner than it would otherwise be. -->
2. [x] If you're not on the contributors list, add yourself to `AUTHORS.md`.
<!-- We'd like to acknowledge your contributions! -->
3. [x] Add a line to `CHANGELOG.md` under the latest version header (i.e. the one that is "on deck") describing the contribution.
    - Do use some discretion here; if there are multiple PRs that are related, keep them in a single line.

# Automatic checks

There will be automatic checks run on the PR. These include:

- Building a preview of the docs on Netlify
- Automatically linting the code
- Making sure the code is documented
- Making sure that all tests are passed
- Making sure that code coverage doesn't go down.

# Relevant Reviewers

<!-- Finally, please tag relevant maintainers to review. -->

Please tag maintainers to review.

- @ericmjl

Commit:
add make_clean_names function that can be applied to polars

Commit:
add examples for make_clean_names

Commit:
changelog

Commit:
limit import location for polars

Commit:
limit import location for polars

Commit:
fix polars in environment-dev.yml

Commit:
install polars in doctest

Commit:
limit polars imports - user should have polars already installed

Commit:
use subprocess.run

Commit:
add subprocess.devnull to docstrings

Commit:
add subprocess.devnull to docstrings

Commit:
add subprocess.devnull to docstrings

Commit:
add subprocess.devnull to docstrings

Commit:
add os.devnull

Commit:
add polars as requirement for docs

Commit:
add polars to tests requirements

Commit:
delete irrelevant folder

Commit:
changelog

Commit:
create submodule for polars

Commit:
fix doctests

Commit:
fix tests; add polars to documentation

Commit:
fix tests; add polars to documentation

Commit:
import janitor.polars

Commit:
control docs output for polars submodule

Commit:
exclude functions in docs rendering

Commit:
exclude functions in docs rendering

Commit:
show_submodules=true

Commit:
fix docstring rendering for polars

Commit:
Expression -> expression

ericmjl committed:
Merge dev into samukweku/polars_clean_names

Commit:
rename functions.py

Commit:
pivot_longer implemented for polars

Commit:
changelog

Commit:
keep changes related only to pivot_longer

Commit:
pd -> pl

Commit:
pd -> pl

Commit:
df.pivot_longer -> df.janitor.pivot_longer

Commit:
df.pivot_longer -> df.janitor.pivot_longer

Commit:
pd -> pl

Commit:
pd -> pl

Commit:
add >>> df

Commit:
add >>> df

Commit:
keep changes related only to polars pivot_longer

Commit:
add polars support to read_commandline

Commit:
remove irrelevant files

ericmjl commented:
<!-- NETLIFY DEPLOY COMMENT GENERATED BY ACTIONS_NETLIFY -->
🚀 Deployed on https://deploy-preview-1356--pyjanitor.netlify.app

Commit:
minor edit to docs

ericmjl committed:
Merge dev into samukweku/polars_read_commandline

ericmjl committed:
Merge dev into samukweku/polars_read_commandline

ericmjl committed:
Merge dev into samukweku/polars_read_commandline

ericmjl committed:
Merge dev into samukweku/polars_read_commandline

ericmjl committed:
Merge dev into samukweku/polars_read_commandline

ericmjl committed:
Merge dev into samukweku/polars_read_commandline

ericmjl reviewed (approved):
Looking great! I am going to see if I can push up a test for `read_commandline()`.

ericmjl committed:
test(io): Parameterize test_read_commandline with engine

This commit modifies the test_read_commandline function in the io module to accept an additional parameter, "engine". This allows the function to be tested with different engines, specifically "pandas" and "polars". The function calls within the test have been updated to pass this new parameter. The change is intended to improve the comprehensiveness of the test coverage.

ericmjl committed:
test(io): update default parameter in test_read_commandline

This commit removes the default value for the 'engine' parameter in the test_read_commandline function. This change ensures that the function's behavior is explicitly defined by the test parameters.

ericmjl commented:
@samukweku we're close with this one -- I think the line that is failing in `test_readcommandline()` is this one under the `polars` engine:

```python
assert df.equals(dataframe)
```

Would you be kind enough to take a look at this particular error? I am quite sure it's got to do with the polars internal API, but that's one I'm not familiar enough with to know.

ericmjl committed:
Merge dev into samukweku/polars_read_commandline

ericmjl committed:
refactor(tests/io): simplify `test_read_commandline` by removing engine parameter

- Removed the `engine` parameter from `test_read_commandline` function to simplify test logic.
- Updated calls to `janitor.io.read_commandline` within the test to not pass the `engine` argument, aligning with the function's updated signature.

This change does not affect the external API, hence no breaking change is introduced.

ericmjl committed:
Merge dev into samukweku/polars_read_commandline

ericmjl committed:
Merge dev into samukweku/polars_read_commandline

ericmjl commented:
Also LGTM, thanks @samukweku!
